### PR TITLE
Ee 6824 

### PR DESCRIPTION
**Pull request by vittoriobadini**
adding cat A non pass test. Refactored tests to be able to use two applicants

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/proving/income/HmrcStub.java [hunk 1]
```diff
@@ -2,120 +2,166 @@
...
 
     private HmrcResponse hmrcResponse = new HmrcResponse();
 
-    void stubPassUser() throws IOException {
+    void stubPassUser(Applicant applicant) throws IOException {
 
-        getMatchedUser();
+        String applicantId = applicant.nino();
+        getMatchedUser(applicant);
+        wireMockRule.stubFor(get(urlMatching("/individuals/employments/paye\\?matchId=" + applicantId + "&fromDate=[0-9\\-]*&toDate=[0-9\\-]*"))
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/HmrcStub.java]: There is a lot of duplication in this class... and the stubs are quite hard to read.  Could we move each stub into a private method?  So for example, this method would look something like this:
```
void stubPassUser(Applicant applicant) throws IOException {
    String applicantId = applicant.nino();
    getMatchedUser(applicant);
    stubPaye(applicantId, "/template/payments/pass/employmentsPayeResponse.json");
    stubIncomePaye(applicantId, "/template/payments/pass/incomePayeResponse.json");
    stubSa(applicantId, "/template/payments/pass/incomeSAResponseEmpty.json");
    stubSaSelfEmployment(applicantId, "/template/payments/pass/incomeSASelfEmploymentsResponse.json");
}
```

File: src/test/java/uk/gov/digital/ho/proving/income/IpsSearchTest.java [hunk 2]
```diff
@@ -119,16 +128,19 @@ public void thatPassingIndividualReturnsSuccess() throws IOException {
...
-                ipsSearchPage.getEmploymentList().containsAll(employers));
+                ipsSearchPage.getEmploymentList().containsAll(applicantEmployers));
     }
 
     @Test
     public void thatCatAPassingIndividualReturnsSuccess() throws IOException {
-        hmrcStub.stubCatAPassUser();
-        employers.add("Acme Inc");
-        submitValidApplicant(employers);
+        applicantEmployers.add("Acme Inc");
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/IpsSearchTest.java:140]: Is this list cleared in between each test?

**Review by mikehalmaho — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/digital/ho/proving/income/HmrcStub.java (hunk 1 above)
Comment (vittoriobadini) [reply src/test/java/uk/gov/digital/ho/proving/income/HmrcStub.java]: 👍

File: src/test/java/uk/gov/digital/ho/proving/income/IpsSearchTest.java (hunk 2 above)
Comment (vittoriobadini) [reply src/test/java/uk/gov/digital/ho/proving/income/IpsSearchTest.java:140]: no is not 😮

Comment (mikehalmaho) [reply src/test/java/uk/gov/digital/ho/proving/income/IpsSearchTest.java:140]: I wonder how it works then :)

**Review by mikehalmaho — APPROVED** [review]

**Pull request merged**